test/functional/apps/discover/_discover.js

```diff
@@ -104,25 +106,176 @@ define(function (require) {
+            return verifyChartData(expectedBarChartData);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show correct data for chart interval Yearly', function () {
+          var chartInterval = 'Yearly';
+          var expectedBarChartData = [ '102.404'];
+          return discoverPage.setChartInterval(chartInterval)
+          .then(function () {
+            return common.sleep(2000);
+          })
+          .then(function () {
+            return verifyChartData(expectedBarChartData);
+          })
+          .catch(common.handleError(this));
+        });
+
+
+
+        bdd.it('should show correct data for chart interval Auto', function () {
+          var chartInterval = 'Auto';
+          var expectedBarChartData = [ '0', '0', '0', '0', '0', '0',
+            '2.7056249999999977', '14.771249999999995', '54.112500000000004',
+            '105.080625', '100.25437500000001', '54.916875', '13.747499999999988',
+            '2.266874999999999', '3.0712500000000063', '14.771249999999995',
+            '49.944374999999994', '99.523125', '103.471875', '51.699375',
+            '12.943124999999995', '1.9743749999999949', '2.3400000000000034',
+            '12.796875', '51.699375', '102.96000000000001', '99.08437500000001',
+            '53.08875', '14.698125000000005', '2.1206249999999898', '0', '0',
+            '0', '0', '0', '0', '0'
+          ];
+          return discoverPage.setChartInterval(chartInterval)
+          .then(function () {
+            return common.sleep(4000);
+          })
+          .then(function () {
+            return verifyChartData(expectedBarChartData);
+          })
+          .catch(common.handleError(this));
+        });
+
+        bdd.it('should show Auto chart interval of 3 hours', function () {
+          var expectedChartInterval = 'by 3 hours';
+          return discoverPage.getChartInterval()
+          .then(function (actualInterval) {
+            expect(actualInterval).to.be(expectedChartInterval);
           })
           .catch(common.handleError(this));
         });
 
 
+        function verifyChartData(expectedBarChartData) {
+          return common.tryForTime(20 * 1000, function tryingForTime() {
+            return discoverPage.getBarChartData()
+            .then(function compareData(paths) {
+              // console.log(paths);
+              // the largest bars are over 100 pixels high so this is less than 1% tolerance
+              var barHeightTolerance = 1;
+              var stringResults = '';
+              var hasFailure = false;
+              for (var y = 0; y < expectedBarChartData.length; y++) {
+                stringResults += y + ': expected = ' + expectedBarChartData[y] + ', actual = ' + paths[y] +
+                 ', Pass = ' + (Math.abs(expectedBarChartData[y] - paths[y]) < barHeightTolerance);
+                if ((Math.abs(expectedBarChartData[y] - paths[y]) > barHeightTolerance)) {
+                  hasFailure = true;
```

Comment:
@jbudz I refactored this code a bit so that it only logs all the comparison data stringResults if there's a failure.  This cleans up the output when the tests pass.  So it is using common.log if there's a failure instead of the common.debug for everything.
I think in another PR I'll probably separate the getBarChartData and a new compareArraysWithTolerance method since a few of the visualize tests do the same thing.